### PR TITLE
gh-124613: Don't run perf tests in JIT builds

### PR DESCRIPTION
**Pull request by pablogsal**
* Issue: gh-124613

### CODE REVIEW

File: Lib/test/test_perf_profiler.py [hunk 1]
```diff
@@ -23,6 +23,17 @@
     raise unittest.SkipTest("test crash randomly on ASAN/MSAN/UBSAN build")
 
 
+def is_jit_build():
+    cflags = sysconfig.get_config_var("PY_CORE_CFLAGS")
+    if not cflags:
+        return False
```
Comment (vstinner) [inline Lib/test/test_perf_profiler.py]:
```suggestion
    cflags = (sysconfig.get_config_var("PY_CORE_CFLAGS") or '')
```

File: Lib/test/test_perf_profiler.py [hunk 2]
```diff
@@ -23,6 +23,17 @@
...
+    return "_Py_JIT" in cflags
+
+
+if is_jit_build():
+    raise unittest.SkipTest("Perf support is not available in jit builds")
```
Comment (vstinner) [inline Lib/test/test_perf_profiler.py]:
```suggestion
    raise unittest.SkipTest("Perf support is not available in JIT builds")
```

**Review by vstinner — APPROVED** [review]
LGTM. I just suggested some minor changes.

**Comment by brandtbucher** [comment]
If we don't want to skip these on most builds, we *could* decorate the tests with `test.support.without_optimizer`. See https://github.com/python/cpython/pull/124794#issuecomment-2384409299.

**Pull request merged**